src/test/java/com/training/services/tests/VoyagesTest.java

```diff
@@ -79,7 +80,32 @@ public void testVoyageShipCodeValidate() {
                 .as(" status is not 200");
         List<Voyages> voyages = voyageResponse.getPayload().getVoyages();
         for (int i = 0; i < voyages.size(); i++) {
-         softAssert.assertThat(voyages.get(i).getShipCode()).isEqualTo("AL");
+            softAssert.assertThat(voyages.get(i).getShipCode()).isEqualTo("AL");
+        }
+    }
+
+    /**
+     * testVoyageMasterSailDateValidate:comparing sailDate with both master1sailDate and master2SailDate.
+     */
+    @Test
+    public void testVoyageMasterSailDateValidate() {
+        Response voyageResponse = getResponse();
+        logger.info("status--->" + voyageResponse.getStatus());
+        List<Voyages> voyages = voyageResponse.getPayload().getVoyages();
+        logger.info("totalVoyagesCount-->" + voyages.size());
+        Assertions.assertThat(voyageResponse.getStatus()).isEqualTo("200")
+                .as(" status is not 200");
+        for (int index = 0; index < voyages.size(); index++) {
+            String sailDate = voyages.get(index).getSailDate();
+            logger.info("sailDate->" + sailDate + " " + "index valuesS1-->" + index);
+            String master1SailDate = voyages.get(index).getMasterSailDate().getMaster1SailDate();
+            logger.info("master1Date->" + master1SailDate + " " + "index valueM1-->" + index);
+            String master2SailDate = voyages.get(index).getMasterSailDate().getMaster2SailDate();
+            logger.info("master2Date->" + master2SailDate + " " + "index valueM2-->" + index);
+            softAssert.assertThat(sailDate.equals(master1SailDate)).isTrue()
+                    .as("sailDate and master1sailDate are not equal");
+            softAssert.assertThat(sailDate.equals(master2SailDate)).isTrue()
+                    .as("sailDate and master2sailDate are not equal");
```

Comment:
Lets use the master1SailDate and master2SailDate directly here instead of defining a new variable.
Also lets use assertThat().equals()